### PR TITLE
Fix: slideover enter animation

### LINKED ISSUE #4883: Modal slide-over enter animation not working
### Package

filament/filament

### Package Version

v2.16.42

### Laravel Version

v9.39.0

### Livewire Version

_No response_

### PHP Version

PHP 8.1.0

### Problem description

Modal slide-over enter animation only fades in opacity.

### Expected behavior

Modal should slide in from the right.

### Steps to reproduce

Create a slide-over table action.

### Reproduction repository

https://github.com/filamentphp/demo

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by awcodes**
Resolves #4883

### CODE REVIEW

**Comment by awcodes** [comment]
Resolves #4883

File: packages/support/resources/views/components/modal/index.blade.php [hunk 1]
```diff
@@ -88,33 +83,38 @@ class="filament-modal {{ $displayClasses }}"
...
             x-ref="modalContainer"
-            x-cloak
             {{ $attributes->class([
-                'relative w-full cursor-pointer pointer-events-none',
+                'relative w-full cursor-pointer pointer-events-none transition',
                 'my-auto p-4' => ! $slideOver,
             ]) }}
         >
             <div
+                x-data="{ show: false }"
```
Comment (zepfietje) [inline packages/support/resources/views/components/modal/index.blade.php:93]:
```suggestion
                x-data="{ isShown: false }"
```

File: packages/support/resources/views/components/modal/index.blade.php [hunk 2]
```diff
@@ -88,33 +83,38 @@ class="filament-modal {{ $displayClasses }}"
...
+                x-show="show"
+                x-init="$nextTick(()=> {
+                    show = isOpen
+                    $watch('isOpen', () => show = isOpen)
+                })"
```
Comment (zepfietje) [inline packages/support/resources/views/components/modal/index.blade.php]:
```suggestion
                x-data="{ isShown: false }"
                x-init="$nextTick(()=> {
                    isShown = isOpen
                    $watch('isOpen', () => isShown = isOpen)
                })"
                x-show="isShown"
```

**Comment by awcodes** [comment]
Looks good to me.

**Comment by zepfietje** [comment]
Cool, I'll still pull these changes in locally and see if it's all fine visually. 👍

**Comment by zepfietje** [comment]
Nice to see a smooth modal animation again! 😄

One issue still though is that sometimes the modal disappears before the hiding animation has completed. Thoughts on how we can fix that?

**Comment by awcodes** [comment]
Different durations maybe?

**Comment by zepfietje** [comment]
Hmm, that'd still be a partial fix probably.
We could also hide the modal and fire the closing event afterwards, maybe?

**Comment by awcodes** [comment]
I guess the problem is that it's two animations that are tied by the same event and I'm not sure of a way to have transitions depend on each other directly. So, I really think it's about getting the timings right.

**Comment by zepfietje** [comment]
Hmm, I tested this by increasing the transition durations, but it doesn't seem like what you suggested.
I think it's rather something to do with the modal content being removed before transition has finished.

**Comment by awcodes** [comment]
Do you have a screen recording by any chance. Either I'm old and just can't see it or I just can't replicate it.

**Comment by danharrin** [comment]
Thanks @awcodes, just gonna merge how it is and Zep can handle any other fixes as required :)

**Comment by zepfietje** [comment]
Was about to suggest this, @danharrin!
The biggest issue has been fixed, so I'll see later if there's a way to polish even further.

**Pull request merged**